draft-ietf-core-problem-details.md

```diff
@@ -168,40 +172,48 @@ Application-specific attributes will be allocated in the
 ## Standard Problem Detail Entries {#new-spdk}
 
 Beyond the Standard Problem Detail keys defined in {{cddl}}, additional
-Standard Problem Detail keys can be registered in the
+Standard Problem Detail keys can be registered for use in the
 `standard-problem-detail-entries` slot (see {{iana-spdk}}).
 
 Standard Problem Detail keys are negative integers, so they never can
-conflict with Custom Problem Detail keys defined for a problem type
+conflict with Custom Problem Detail keys defined for a specific
+application domain
 (which are unsigned integers or URIs.)
 
 In summary, the keys for Standard Problem Detail entries are in a
 global namespace that is not specific to a particular application domain.
 
 Consumers of a Concise Problem Details instance MUST ignore any Standard Problem
-Detail entries that they do not recognize; this allows problem types to evolve.
+Detail entries that they do not recognize; this allows problem details to evolve.
 
 ## Custom Problem Detail Entries {#new-cpdk}
 
-Applications may extend the Problem Details document with
+Applications may extend the Problem Details data item with
 additional entries to convey additional, application-specific information.
 
 Such new entries are allocated in the `custom-problem-detail-entries` slot, and
-are nested into a map specific to that application.  The map key can either be
-a URI (controlled by the entity defining this extension), or an unsigned int.
-Only the latter needs to be registered ({{iana-cpdk}}).  Within the nested map
-any number of attributes can exist.  When the extension is registered (i.e.,
-unsigned int) the semantics of each custom attribute MUST be described in the
-documentation that goes along with the registration.
-
-The unsigned integer form allows a more compact representation but authors are
-expected to comply with the required registration and documentation process.
+carry a nested map specific to that application.  The map key can either be
+an (absolute!) URI (controlled by the entity defining this extension),
+or an unsigned integer.
+Only the latter needs to be registered ({{iana-cpdk}}).
+
+Within the nested map any number of attributes can be given for a
+single extension.
+The semantics of each custom attribute MUST be described in the
```

Comment:
ISTM that this MUST is only enforceable for registered (uint) keys where the registration procedure imposes the check.

Reply:
Indeed, but interoperability requirements don't need to be "enforceable"...

Reply:
Sometimes one needs to state the obvious 🤣